onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -856,6 +874,24 @@ def _infer_Pool(self, node):
                 helper.make_tensor_value_info(o, vi.type.tensor_type.elem_type,
                                               get_shape_from_sympy_shape(sympy_shape)))
 
+    def _infer_BatchNormalization(self, node):
```

Comment:
>_infer_BatchNormalization [](start = 8, length = 25)

why do we need to add shape inference for BatchNormalization? It should fallback to onnx's shape inference for this op. Is there a bug in onnx implementation for BatchNormalization?

Reply:
Yes, shape inference for outputs 1-4 has only been implemented for opset 14, the older versions others don't have it yet: https://github.com/onnx/onnx/blob/master/onnx/defs/nn/old.cc#L1733